src/index.ts

```diff
@@ -248,21 +251,23 @@ class Compiler {
   }
 
   private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
-    if (!rule.type) rule.type = 'string'
+    if (rule.type === undefined) rule.type = 'string'
```

Comment:
Note: `!rule.type` returns `false` if `rule.type === 'boolean'` or any string value. I think the correct intent is `rule.type === undefined`